.storybook/decorators/PropCombinations/PropCombinationsDecorator.tsx

```diff
@@ -0,0 +1,97 @@
+/**
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ CAUTION ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * Updating this file will flag a change
+ * in _every_ component that leverages generated stories.
+ *
+ * Modify with caution
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ */
+
+import React from 'react';
+import { Decorator, StoryContext, StoryFn } from '@storybook/react';
+import { entries, isUndefined } from 'lodash';
+
+import {
+  StoryType,
+  type GeneratedStoryConfig,
+  type StoryMetaType,
+} from '@leafygreen-ui/lib';
+
+import { generatedStoryWrapper } from './PropCombinations.styles';
+import { Err, PropCombinations } from './components';
+import { isGeneratedStory } from './utils';
+import { PARAM_NAME } from './constants';
+
+type ContextType<T extends React.ComponentType<any>> = StoryContext<T> &
+  StoryMetaType<T>;
+
+const PropCombinationsDecorator: Decorator = (
```

Comment:
I know this is in reference to Chromatic's application of decorators, but I would expect this to be used at some point as a decorator as defined by [TS](https://www.typescriptlang.org/docs/handbook/decorators.html) or [JS](https://github.com/tc39/proposal-decorators) specs/documentation. Can we address this somehow in the documentation to set folks who are new to this functionality, but familiar with the language construct, up for success?

As is, I would expect to be able to prefix something in the story file with `@PropCombinationsDecorator()` to use this (without deep-diving into Storybook's usage of decorators).

Reply:
Is this just a comment on the naming convention? `PropCombinationsStorybookDecorator`?